### PR TITLE
Refactor TabBarBadgeCounts to Core

### PR DESCRIPTION
**Pull request by vanwagonet**
Removes ReactiveSwift from being used for badges

refs: none
affects: none
release note: none

### CODE REVIEW

File: rn/Teacher/ios/Teacher/RootTabBarController.swift [hunk 1]
```diff
@@ -81,24 +70,10 @@ class RootTabBarController: UITabBarController {
...
-extension RootTabBarController: UITabBarControllerDelegate {
-    func tabBarController(_ tabBarController: UITabBarController, animationControllerForTransitionFrom fromVC: UIViewController, to toVC: UIViewController) -> UIViewControllerAnimatedTransitioning? {
-        return NoAnimatedTransitioning()
-    }
-
-    func tabBarController(_ tabBarController: UITabBarController, shouldSelect viewController: UIViewController) -> Bool {
-        tabBarController.resetViewControllerIfSelected(viewController)
-        return true
-    }
-}
```
Comment (vanwagonet) [inline rn/Teacher/ios/Teacher/RootTabBarController.swift:104]: this "fix" wasn't in Student and I can't reproduce any issues related when removing this.

**Comment by vanwagonet** [comment]
test plan:
- switch users or read an inbox message or otherwise do things that should change the counts in Student & Teacher.
- verify that the tab bar badges and the application badge are updated.

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/9r83rbd) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2F9r83rbd)
teacher: [Link](https://rebrand.ly/wbda98w) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fwbda98w)
parent: [Link](https://rebrand.ly/oxf1mn7) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Foxf1mn7)

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: None

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 93.61% | 93.12% | 0.49%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Pull request merged**